extensions/lifecycle/tasks/LifecycleColdStatusArchiveTask.js

```diff
@@ -0,0 +1,75 @@
+const async = require('async');
+
+const ObjectMDArchive = require('arsenal').models.ObjectMDArchive;
+const LifecycleUpdateTransitionTask = require('./LifecycleUpdateTransitionTask');
+
+class LifecycleColdStatusArchiveTask extends LifecycleUpdateTransitionTask {
+    /**
+     * Process a lifecycle object entry
+     *
+     * @constructor
+     * @param {LifecycleObjectProcessor} proc - object processor instance
+     */
+    constructor(proc) {
+        const procState = proc.getStateVars();
+        super();
+        Object.assign(this, procState);
+    }
+
+    getTargetAttribute(entry) {
+        const {
+            bucketName,
+            objectKey,
+            objectVersion,
+            accountId
+        } = entry.getTarget();
+        return {
+            bucket: bucketName,
+            key: objectKey,
+            version: objectVersion,
+            accountId
+        };
+    }
+
+    getSourceAttribute(entry) {
+        const { bucketName, objectKey, objectVersion } = entry.getTarget();
+        return {
+            bucket: bucketName,
+            objectKey,
+            version: objectVersion,
+            storageClass: entry.get('sourceStorageClass'),
+            lastModified: entry.get('sourceLastModified'),
+        };
+    }
+
+    _updateMdWithArchiveInfo(entry, objMD) {
```

Comment:
Should probably mention in the name that it clears all archive attributes except archiveinfo